src/stlink-lib/chipid.c

```diff
@@ -1121,11 +1122,45 @@ void init_chipids(char *dir_to_scan)
       dump_a_chip (stderr, op);
       fprintf (stderr, "---------- new ------------\n");
       dump_a_chip (stderr, p);
-      
+
     }
   }
 #endif
 }
+#endif //STLINK_HAVE_DIRENT_H
+
+#if defined(_WIN32) && !defined(STLINK_HAVE_DIRENT_H)
+#include <fileapi.h>
+#include <strsafe.h>
+void init_chipids(char *dir_to_scan)
+{
+  HANDLE hFind = INVALID_HANDLE_VALUE;
+  WIN32_FIND_DATAA ffd;
+  char file_pattern[MAX_PATH] = {0};
+  char filepath[MAX_PATH] = {0};
+  StringCchCopyA(file_pattern, STLINK_ARRAY_SIZE(file_pattern), dir_to_scan);
+  if (FAILED(StringCchCatA(file_pattern, STLINK_ARRAY_SIZE(file_pattern), "\\*.chip"))) {
+    ELOG("Path to chips's dir too long.\n");
+    return;
+  };
+  hFind = FindFirstFileA(file_pattern, &ffd);
+  if (INVALID_HANDLE_VALUE == hFind){
+    ELOG("Can't find any chip description file in %s.\n", file_pattern);
+    return;
+  }
+
+  do {
+      memset(filepath, 0, STLINK_ARRAY_SIZE(filepath));
+      StringCchCopyA(filepath, STLINK_ARRAY_SIZE(filepath), dir_to_scan);
+      StringCchCatA(filepath, STLINK_ARRAY_SIZE(file_pattern), "\\");
+      StringCchCatA(filepath, STLINK_ARRAY_SIZE(file_pattern), ffd.cFileName);
+      process_chipfile(filepath);
```

Comment:
Too deep indentation.

Reply:
This should be 4 empty spaces only.

Reply:
according to this https://github.com/stlink-org/stlink/blob/develop/CONTRIBUTING.md#coding-style
there should be 8 spaces.

Reply:
Yes, but in the current patch version there seem to be 6 spaces. So you should either follow the style of the rest of this file, and use 2+2=4 spaces, or follow that guide, and use 4+4=8 spaces.

Reply:
Well, true actually, because do { } is already indented...